workers/generic-worker/payload_test.go

```diff
@@ -146,6 +146,21 @@ func NowMillis(t *testing.T) (now time.Time) {
 	return
 }
 
+// If a task deadline is sooner than maxRunTime
+func TestTaskDeadlineTooClose(t *testing.T) {
+	now := NowMillis(t)
+	task := taskWithPayload(`{
+  "env": {
+    "XPI_NAME": "dist/example_add-on-0.0.1.zip"
+  },
+  "maxRunTime": 3,
+  "command": [` + rawHelloGoodbye() + `]
+}`)
+	task.Definition.Deadline = tcclient.Time(now.Add(time.Second))
```

Comment:
This could put deadline in the past by the time the task gets claimed, especially if not run against mocks but a real service, so I think better to set it for e.g. 10 mins in the future.